tests/common/test_version.py

```diff
@@ -113,39 +111,45 @@ def test_distro_is_correct_for_exception_case(self, *args): # pylint: disable=us
             osinfo = get_distro()
 
         self.assertListEqual(default_list, osinfo)
-        return
+
+    def test_get_lis_version_should_return_a_string(self):
+        """
+        On a Hyper-V guest, this function should return a string of the
+        version, like '4.3.5'. Anywhere else it should return
+        'Absent', and should not return 'Failed', so we check that it
+        returns a string but not 'Failed'.
+        """
+        lis_version = get_lis_version()
+        self.assertIsInstance(lis_version, str)
+        self.assertNotEqual(lis_version, "Failed")
```

Comment:
how is the test ensuring that the function will never return "Failed"? We run unit tests in a variety of platforms (OS X, Linux, Windows, physical machines, VMs, containers)

Reply:
I don't understand your question. It will only return `Failed` if an exception is raised when calling `run_command`. As I asked you above, I'm not sure how you want that handled, so I went with catching the exception and returning "Failed." As that should never be the case, this test should pass on all those platforms. If the test fails, something weird has happened that needs debugging (which is the point of the assertion).

Reply:
(Also I've updated the doc strings to be more clear about this.)

Reply:
My concern here is whether this assert will ever fail and force us to break our development cycle to debug the failure.

The production code doesn't seem to care at all about the failure. Why does the test do? If you are trying to get a notification of the failure you would be better handling this in the production code with, for example, a telemetry event that you guys would monitor.

Reply:
> The production code doesn't seem to care at all about the failure.

This is correct, but wouldn't you be? I was trying to write as extensive of a unit test as possible (given the contributor guidelines) especially where it does not make sense to mock the function.

I'll remove the assertion for you.